From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Wed, 26 Aug 2026 11:56:25 +0200
Subject: fix(systemd-cryptsetup): support grep applet from busybox in
 module-setup.sh

busybox's `grep` applet does not support the Perl regexp `-P`.

Use the extended regexp `-E` instead. This does not support the inline
Perl `(?i)` flag. So make the entire pattern case-insensitive.

Part of: https://github.com/dracut-ng/dracut/issues/2437
Origin: upstream, https://github.com/dracut-ng/dracut/commit/a2a8289052e9b94e1b28f945d65a4b3d9deb362f
---
 modules.d/71systemd-cryptsetup/module-setup.sh | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/modules.d/71systemd-cryptsetup/module-setup.sh b/modules.d/71systemd-cryptsetup/module-setup.sh
index 4bad1ef..66b35ed 100755
--- a/modules.d/71systemd-cryptsetup/module-setup.sh
+++ b/modules.d/71systemd-cryptsetup/module-setup.sh
@@ -92,7 +92,7 @@ install() {
 
                 if [ -z "$service_name" ]; then
                     # if no explicit Service= is defined, construct the service name based on the socket unit's name
-                    if grep -P -q "^Accept\s*=\s*(?i)(1|yes|y|true|t|on)$" "$socket_unit"; then
+                    if grep -Eiq "^Accept\s*=\s*(1|yes|y|true|t|on)$" "$socket_unit"; then
                         # if Accept is truthy, assemble a service template
                         service_name=$(basename "$socket_unit" .socket)"@.service"
                     else
@@ -111,12 +111,12 @@ install() {
                 # which itself depends on cryptsetup.target. This could lead to either:
                 #   a) systemd-cryptsetup falling back to a passphrase prompt due to a missing socket file
                 #   b) a deadlock caused by a circular dependency (service unit -> sysinit.target -> cryptsetup.target -> service unit)
-                if ! grep -P -q "^DefaultDependencies\s*=\s*(?i)(0|no|n|false|f|off)" "$socket_unit"; then
+                if ! grep -Eiq "^DefaultDependencies\s*=\s*(0|no|n|false|f|off)" "$socket_unit"; then
                     dwarning "crypt: $socket_unit: default dependencies are not disabled," \
                         "the socket file may not exist by the time systemd-cryptsetup gets executed"
                 fi
 
-                if ! grep -P -q "^DefaultDependencies\s*=\s*(?i)(0|no|n|false|f|off)" "${socket_unit%/*}/$service_name"; then
+                if ! grep -Eiq "^DefaultDependencies\s*=\s*(0|no|n|false|f|off)" "${socket_unit%/*}/$service_name"; then
                     dwarning "crypt: ${socket_unit%/*}/$service_name: default dependencies are not disabled," \
                         "the service unit may encounter a deadlock due to a circular dependency"
                 fi
